src/simulation.cpp

```diff
@@ -464,17 +469,40 @@ void finalize_batch()
   }
 
   // Write out surface source if requested.
-  if (settings::surf_source_write &&
-      simulation::current_batch == settings::n_batches) {
-    auto filename = settings::path_output + "surface_source";
-    auto surf_work_index =
-      mpi::calculate_parallel_index_vector(simulation::surf_source_bank.size());
-    gsl::span<SourceSite> surfbankspan(simulation::surf_source_bank.begin(),
-      simulation::surf_source_bank.size());
-    if (settings::surf_mcpl_write) {
-      write_mcpl_source_point(filename.c_str(), surfbankspan, surf_work_index);
-    } else {
-      write_source_point(filename.c_str(), surfbankspan, surf_work_index);
+  if (settings::surf_source_write) {
+    if (simulation::current_batch == settings::n_batches &&
+        settings::split_file_per_batch == false) {
+      auto filename = settings::path_output + "surface_source";
+      auto surf_work_index = mpi::calculate_parallel_index_vector(
+        simulation::surf_source_bank.size());
+      gsl::span<SourceSite> surfbankspan(simulation::surf_source_bank.begin(),
+        simulation::surf_source_bank.size());
+      if (settings::surf_mcpl_write) {
+        write_mcpl_source_point(
+          filename.c_str(), surfbankspan, surf_work_index);
+      } else {
+        write_source_point(filename.c_str(), surfbankspan, surf_work_index);
+      }
+    } else if (settings::split_file_per_batch &&
+               simulation::current_batch >
+                 (settings::n_batches - settings::n_surf_source_files)) {
+      auto filename = settings::path_output + "surface_source_batch_" +
```

Comment:
It would probably be best to follow the same naming convention as statepoint files here 
(i.e. `surface_source.<batch_number>.h5`)